nixnet/database/_frame.py

```diff
@@ -239,28 +243,24 @@ def mux_subframes(self):
         return self._mux_subframes
 
     @property
-    def pdu_refs(self):
-        return _props.get_frame_pdu_refs(self._handle)
-
-    @pdu_refs.setter
-    def pdu_refs(self, value):
-        _props.set_frame_pdu_refs(self._handle, value)
-
-    @property
-    def pdu_start_bits(self):
-        return _props.get_frame_pdu_start_bits(self._handle)
-
-    @pdu_start_bits.setter
-    def pdu_start_bits(self, value):
-        _props.set_frame_pdu_start_bits(self._handle, value)
-
-    @property
-    def pdu_update_bits(self):
-        return _props.get_frame_pdu_update_bits(self._handle)
-
-    @pdu_update_bits.setter
-    def pdu_update_bits(self, value):
-        _props.set_frame_pdu_update_bits(self._handle, value)
+    def pdus(self):
+        handles = [_pdu.Pdu(handle)
+                   for handle in _props.get_frame_pdu_refs(self._handle)]
+        pdu_tuples = zip(*(handles,
+                           _props.get_frame_pdu_start_bits(self._handle),
+                           _props.get_frame_pdu_update_bits(self._handle)))
+        for pdu in [types.PduFrameReference(ref, start_bit, update_bit)
+                    for (ref, start_bit, update_bit) in pdu_tuples]:
+            yield pdu
```

Comment:
Why create a list, then iterate over it, yielding the results?  That feels overly complicated.

Instead you could just
```python
return (
    types.PduFrameReference(ref, start_bit, update_bit)
    for (ref, start_bit, update_bit) in pdu_tuples
)
```
or
```python
for (ref, start_bit, update_bit) in pdu_tuples]:
    yield types.PduFrameReference(ref, start_bit, update_bit)
```

Reply:
Yeah, the last one looks cleaner to me too.